src/components/controller.tsx

```diff
@@ -30,16 +30,14 @@ export default function Controller() {
             filename={filename}
             savedata={savedata}
           />
-          <Editor savedata={savedata} setSavedata={setSavedata} />
+          <Editor savedata={savedata} />
```

Comment:
`setSavedata` was not needed because `Editor` works directly on the XML so the reference is update without the need to call `setSavedata`

Reply:
I might've used `setSavedata` to force update components when the XML changes. I know it's very bad React code. I wrote a long time ago.

Ideally, we'd be using `useSyncExternalStore` to udpdate the UI whenever the XML document changes, but I never took the time to do that refactor.

Reply:
Gotcha, would you like that I add a todo comment mentioning a refactor using `useSynExternalStore`?

Reply:
No, it's fine as is. Looks to be working correctly.